### PR TITLE
Release: 3.1.0

### PR DESCRIPTION
**Pull request by StarlightIbuki**
Notice that the changelog is edited by hand. Please review the changelog, especially the brief.

**NOTICE: PLEASE do not squash the commits! The commits are already squashed once. The other 2 commits are supposed to be in the git history.**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,18 +65,19 @@
 - [0.10.0](#0100---20170307)
 - [0.9.9 and prior](#099---20170202)
 
+## [3.1.0]
 
-## Unreleased
+> Released 2022/12/02
```
Comment (tyler-ball) [inline CHANGELOG.md:70]:
```suggestion
> Released 2022/12/05
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -64,18 +65,19 @@
...
-#### Core
-
-- Change the reponse body for a TRACE method from `The upstream server responded with 405`
-  to `Method not allowed`, make the reponse to show more clearly that Kong do not support
-  TRACE method.
-  [#9448](https://github.com/Kong/kong/pull/9448)
+This major release introduces more flexible management of keys, dynamic log
+level, fixed dataplane ID, HTTP forward proxy support, and other features
+to help ease management. Various plugins are also extended.
+Internal changes are also made to improve Kong's performance.
```
Comment (tyler-ball) [inline CHANGELOG.md:75]:
```suggestion
Internal changes are made to improve Kong's performance.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -64,18 +65,19 @@
...
+As it is a major release, users are advised
```
Comment (tyler-ball) [inline CHANGELOG.md:76]:
```suggestion
As it is a significant release, users are advised
```

`major` has a specific SemVer connotation, lets avoid it. Also do we have any breaking changes in a minor release?

**Review by tyler-ball — APPROVED** [review]
Are these changes also needed in EE?

File: CHANGELOG.md (hunk 3 above)
Comment (StarlightIbuki) [reply CHANGELOG.md]: @tyler-ball Could you review the changelog again? I'm not very confident that I understand the changes right.

**Review by tyler-ball — APPROVED** [review]

**Pull request merged**